src/huggingface_hub/inference_api.py

```diff
@@ -0,0 +1,108 @@
+import json
+import logging
+from typing import Optional
+
+import requests
+
+from .hf_api import HfApi, HfFolder
+from .validation import ALL_TASKS, check_inputs, check_params
+
+
+logger = logging.getLogger(__name__)
+
+
+ENDPOINT = "https://api-inference.huggingface.co/pipeline"
```

Comment:
This endpoint is still not 100% supported/documented If we can avoid putting it in a library that's better.

It's not documented because we have no way currently to refuse loading invalid pipelines.

Reply:
I prefer giving users the flexibility to override which pipeline is being used. If the pipeline from the repository != override pipeline, this will give a warning (`"You're using a different task than the one specified in the repository. Be sure to know what you're doing :)"`). I can add a commit to improve this warning, but I think we should let users have a bit more control.

Reply:
I agree, and it makes more sense in the broader sense of the information flow (we query the hub to get the task).
So it's not a strong opposition, but if we go through with this PR, I will add and commit to supporting the `pipeline` endpoint.

Reply:
IMO we should still expose the "standard" endpoint in the code (without the `pipeline/` part of the URL) as this documents the fact that it is the endpoint that will get called 95% of the time.